synapse/lib/lmdblayer.py

```diff
@@ -52,18 +54,24 @@ class LmdbLayer(s_layer.Layer):
         readahead = self.conf.get('lmdb:readahead')
         maxsize = self.conf.get('lmdb:maxsize')
         growsize = self.conf.get('lmdb:growsize')
+
+        map_async = core.conf.get('layer:lmdb:map_async')
+        self.conf.setdefault('lmdb:map_async', map_async)
+
+        map_async = self.conf.get('lmdb:map_async')
+
         # First check hive configuration.  If not set, use passed-in parameter (that defaults to False)
         self.lockmemory = self.conf.get('lmdb:lockmemory')
         if self.lockmemory is None:
             self.lockmemory = core.conf.get('dedicated')
 
         self.layrslab = await s_lmdbslab.Slab.anit(path, max_dbs=128, map_size=mapsize, maxsize=maxsize,
                                                    growsize=growsize, writemap=True, readahead=readahead,
-                                                   lockmemory=self.lockmemory)
+                                                   lockmemory=self.lockmemory, map_async=map_async)
         self.onfini(self.layrslab.fini)
 
         self.spliceslab = await s_lmdbslab.Slab.anit(splicepath, max_dbs=128, map_size=mapsize, maxsize=maxsize,
-                                                     growsize=growsize, writemap=True, readahead=readahead)
+                                                     growsize=growsize, writemap=True, readahead=readahead, map_async=map_async)
```

Comment:
in _onCoFini do we need to call `self.lenv.sync(force=True)` during teardown if this option is set?

Reply:
nope, he def does the sync for us :)